gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java

```diff
@@ -62,41 +62,54 @@ public FileSystem getFileSystem() {
     return this.fs;
   }
 
+  private void updateFileSystem(String uri) throws IOException, InterruptedException, URISyntaxException {
+
+    if (state.getPropAsBoolean(ConfigurationKeys.SHOULD_FS_PROXY_AS_USER,
+        ConfigurationKeys.DEFAULT_SHOULD_FS_PROXY_AS_USER)) {
+      // Initialize file system as a proxy user.
+      this.fs =
+          new ProxiedFileSystemWrapper().getProxiedFileSystem(state, ProxiedFileSystemWrapper.AuthType.TOKEN,
+              state.getProp(ConfigurationKeys.FS_PROXY_AS_USER_TOKEN_FILE), uri);
+
+    } else {
+      // Initialize file system as the current user.
+      this.fs = FileSystem.get(URI.create(uri), this.configuration);
+    }
+  }
+
   @Override
-  public void connect()
-      throws FileBasedHelperException {
+  public void connect() throws FileBasedHelperException {
     URI uri = null;
     try {
-      uri = new URI(state.getProp(ConfigurationKeys.SOURCE_FILEBASED_FS_URI));
-      this.fs = FileSystem.get(uri, configuration);
+      this.updateFileSystem(state.getProp(ConfigurationKeys.SOURCE_FILEBASED_FS_URI));
     } catch (IOException e) {
       throw new FileBasedHelperException("Cannot connect to given URI " + uri + " due to " + e.getMessage(), e);
     } catch (URISyntaxException e) {
       throw new FileBasedHelperException("Malformed uri " + uri + " due to " + e.getMessage(), e);
+    } catch (InterruptedException e) {
+      throw new FileBasedHelperException("Interruptted exception is caught when getting the proxy file system", e);
     }
   }
 
   @Override
-  public void close()
-      throws FileBasedHelperException {
-        /*
-         * TODO
-         * Removing this for now, FileSystem.get() returns the same object each time it is called within a process
-         * If this method gets called in parallel across tasks it is going to cause problems
-         * Basically, close cannot be called multiple times within a process
-         * http://stackoverflow.com/questions/17421218/multiples-hadoop-filesystem-instances
-         */
-
-//        try {
-//            this.fs.close();
-//        } catch (IOException e) {
-//            throw new FileBasedHelperException("Cannot close Hadoop filesystem due to "  + e.getMessage(), e);
-//        }
+  public void close() throws FileBasedHelperException {
+    /*
+     * TODO
+     * Removing this for now, FileSystem.get() returns the same object each time it is called within a process
+     * If this method gets called in parallel across tasks it is going to cause problems
+     * Basically, close cannot be called multiple times within a process
+     * http://stackoverflow.com/questions/17421218/multiples-hadoop-filesystem-instances
+     */
+
+    //        try {
```

Comment:
Please remove those lines as this is exactly what git provides: maintain code history so you won't lose it.